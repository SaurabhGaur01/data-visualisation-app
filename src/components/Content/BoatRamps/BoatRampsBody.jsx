import React from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { compose } from "redux";
import TableBody from "@material-ui/core/TableBody";
import TableCell from "@material-ui/core/TableCell";
import TableRow from "@material-ui/core/TableRow";
import handleMaterial from "../../../thunks/handleMaterial";

const BoatRampsBody = ({
  filteredFeatures,
  actionHandleMaterial,
  rowsPerPage,
  page,
}) => {
  const handleClick = (event, name) => {
    actionHandleMaterial(name.material);
  };

  return (
    <TableBody>
      {filteredFeatures
        .slice(page * rowsPerPage, page * rowsPerPage + rowsPerPage)
        .map((row) => (
          <TableRow
            hover
            onClick={(event) => handleClick(event, row)}
            role="checkbox"
            tabIndex={-1}
            key={row.rowID}
          >
            <TableCell align="right">{row.material}</TableCell>
            <TableCell align="right">{row.area}</TableCell>
            <TableCell align="right">{row.status}</TableCell>
            <TableCell align="right">{row.assetNumb}</TableCell>
            <TableCell align="right">{row.owner}</TableCell>
          </TableRow>
        ))}
    </TableBody>
  );
};

BoatRampsBody.propTypes = {
  filteredFeatures: PropTypes.arrayOf(PropTypes.shape({})),
  actionHandleMaterial: PropTypes.func.isRequired,
  page: PropTypes.number.isRequired,
  rowsPerPage: PropTypes.number.isRequired,
};

export const mapStateToProps = (state) => ({
  filteredFeatures: state.filteredFeatures,
});

const mapDispatchAsProps = {
  actionHandleMaterial: handleMaterial,
};

const hocChain = compose(
    connect(mapStateToProps, mapDispatchAsProps)
);

export { BoatRampsBody as TestableBoatRampsBody };

export default hocChain(BoatRampsBody);
