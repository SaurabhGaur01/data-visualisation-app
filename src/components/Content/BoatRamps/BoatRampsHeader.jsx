import React from "react";
import TableCell from "@material-ui/core/TableCell";
import TableHead from "@material-ui/core/TableHead";
import TableRow from "@material-ui/core/TableRow";
import TableSortLabel from "@material-ui/core/TableSortLabel";

const headCells = [
  { id: "material", label: "Material" },
  { id: "area", label: "Area" },
  { id: "status", label: "Status" },
  { id: "assetNumb", label: "Asset Number" },
  { id: "owner", label: "Owner" },
];

const BoatRampsHeader = () => (
  <TableHead>
    <TableRow>
      {headCells.map(({ id, label }) => (
        <TableCell key={id} align="right">
          <TableSortLabel>{label}</TableSortLabel>
        </TableCell>
      ))}
    </TableRow>
  </TableHead>
);

export default BoatRampsHeader;
