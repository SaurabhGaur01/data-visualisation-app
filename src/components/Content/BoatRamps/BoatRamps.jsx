import React from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { compose } from "redux";
import Table from "@material-ui/core/Table";
import TableContainer from "@material-ui/core/TableContainer";
import TablePagination from "@material-ui/core/TablePagination";
import Paper from "@material-ui/core/Paper";
import BoatRampsHeader from "./BoatRampsHeader";
import BoatRampsBody from "./BoatRampsBody";

const BoatRamps = ({ filteredFeatures }) => {
  const [page, setPage] = React.useState(0);
  const [rowsPerPage, setRowsPerPage] = React.useState(10);

  const handleChangePage = (event, newPage) => {
    setPage(newPage);
  };

  const handleChangeRowsPerPage = (event) => {
    setRowsPerPage(parseInt(event.target.value, 10));
    setPage(0);
  };

  return (
    <Paper className="mt-3">
      <TableContainer>
        <Table
          className="table-content"
          aria-labelledby="tableTitle"
          size="small"
          aria-label="enhanced table"
        >
          <BoatRampsHeader />
          <BoatRampsBody page={page} rowsPerPage={rowsPerPage} />
        </Table>
        <TablePagination
          rowsPerPageOptions={[10, 25]}
          component="div"
          count={filteredFeatures.length}
          rowsPerPage={rowsPerPage}
          page={page}
          onChangePage={handleChangePage}
          onChangeRowsPerPage={handleChangeRowsPerPage}
        />
      </TableContainer>
    </Paper>
  );
};

BoatRamps.propTypes = {
  filteredFeatures: PropTypes.arrayOf(PropTypes.shape({})),
};

export const mapStateToProps = (state) => ({
  filteredFeatures: state.filteredFeatures,
});

const hocChain = compose(
    connect(mapStateToProps)
);

export { BoatRamps as TestableBoatRamps };

export default hocChain(BoatRamps);
