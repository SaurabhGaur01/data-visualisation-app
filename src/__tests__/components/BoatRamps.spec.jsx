import React from 'react';
import { shallow } from 'enzyme';
import { TestableBoatRamps, mapStateToProps} from '../../components/Content/BoatRamps/BoatRamps';

describe('BoatRamps Component', () => {
    let props;

    beforeEach(() =>{
        props = {
            filteredFeatures: [
                {
                    rowID: 1,
                    type: "mockType",
                    material: "mockMaterial",
                    area: 150.25,
                    status: "mockStatus",
                    assetNumb: "mockAsset",
                    owner: "mockOwner",
                },
            ],
        };
    });

    it('should render correctly', () => {
        const renderedModule = shallow(<TestableBoatRamps {...props} />);
        expect(renderedModule).toMatchSnapshot();
    });

    describe('mapStateToProps', () => {    
        const mockFilteredMaterial = [{ mockKey: 'mockValue'}]    
        const mockedState = {
            filteredFeatures: mockFilteredMaterial
        };

        it('should return the module state from the redux store', () => {
            const mockPropsFromRedux = mapStateToProps(mockedState);
            expect(mockPropsFromRedux).toEqual(
                {
                    filteredFeatures: mockFilteredMaterial,
                }
            )
        })
    })
});    