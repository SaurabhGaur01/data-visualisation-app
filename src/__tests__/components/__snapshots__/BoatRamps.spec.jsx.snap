// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BoatRamps Component should render correctly 1`] = `
<WithStyles(ForwardRef(Paper))
  className="mt-3"
>
  <WithStyles(ForwardRef(TableContainer))>
    <WithStyles(ForwardRef(Table))
      aria-label="enhanced table"
      aria-labelledby="tableTitle"
      className="table-content"
      size="small"
    >
      <BoatRampsHeader />
      <Connect(BoatRampsBody)
        page={0}
        rowsPerPage={10}
      />
    </WithStyles(ForwardRef(Table))>
    <WithStyles(ForwardRef(TablePagination))
      component="div"
      count={1}
      onChangePage={[Function]}
      onChangeRowsPerPage={[Function]}
      page={0}
      rowsPerPage={10}
      rowsPerPageOptions={
        Array [
          10,
          25,
        ]
      }
    />
  </WithStyles(ForwardRef(TableContainer))>
</WithStyles(ForwardRef(Paper))>
`;
